BUILD.bazel

```diff
@@ -0,0 +1,151 @@
+load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package")
+load(":rollup.bzl", "rollup")
+
+exports_files([
+    "tsconfig.json",
+    "tsconfig-node.json",
+    "angular_license.txt",
+])
+
+ES5_BUNDLES = {
+    "async-test": "lib/testing/async-testing",
+    "fake-async-test": "lib/testing/fake-async",
+    "long-stack-trace-zone": "lib/zone-spec/long-stack-trace",
+    "proxy": "lib/zone-spec/proxy",
+    "zone-patch-rxjs-fake-async": "lib/rxjs/rxjs-fake-async",
+    "sync-test": "lib/zone-spec/sync-test",
+    "task-tracking": "lib/zone-spec/task-tracking",
+    "wtf": "lib/zone-spec/wtf",
+    "zone": "lib/browser/rollup-legacy-main",
+    "zone-error": "lib/common/error-rewrite",
+    "zone-legacy": "lib/browser/browser-legacy",
+    "zone-mix": "lib/mix/rollup-mix",
+    "zone-node": "lib/node/rollup-main",
+    "zone-bluebird": "lib/extra/bluebird",
+    "zone-patch-canvas": "lib/browser/canvas",
+    "zone-patch-cordova": "lib/extra/cordova",
+    "zone-patch-electron": "lib/extra/electron",
+    "zone-patch-fetch": "lib/common/fetch",
+    "jasmine-patch": "lib/jasmine/jasmine",
+    "zone-patch-jsonp": "lib/extra/jsonp",
+    "webapis-media-query": "lib/browser/webapis-media-query",
+    "mocha-patch": "lib/mocha/mocha",
+    "webapis-notification": "lib/browser/webapis-notification",
+    "zone-patch-promise-test": "lib/testing/promise-testing",
+    "zone-patch-resize-observer": "lib/browser/webapis-resize-observer",
+    "webapis-rtc-peer-connection": "lib/browser/webapis-rtc-peer-connection",
+    "zone-patch-rxjs": "lib/rxjs/rxjs",
+    "webapis-shadydom": "lib/browser/shadydom",
+    "zone-patch-socket-io": "lib/extra/socket-io",
+    "zone-patch-user-media": "lib/browser/webapis-user-media",
+    "zone-testing": "lib/testing/zone-testing",
+    "zone-testing-bundle": "lib/browser/rollup-legacy-test-main",
+    "zone-testing-node-bundle": "lib/node/rollup-test-main",
+}
+
+ES2015_BUNDLES = {
+    "zone-evergreen": "lib/browser/rollup-main",
+    "zone-evergreen-testing-bundle": "lib/browser/rollup-test-main",
+}
+
+# the rollup bundles
+rollup(ES5_BUNDLES)
+rollup(ES2015_BUNDLES)
+
+# the es5 filegroups
+[
+    filegroup(
+        name = b + ".es5",
+        srcs = [":" + b],
+        output_group = "es5_umd",
+    )
+    for b in ES5_BUNDLES
+]
+
+# the es5.min filegroups
+[
+    filegroup(
+        name = b + ".es5.min",
+        srcs = [":" + b],
+        output_group = "es5_umd_min",
+    )
+    for b in ES5_BUNDLES
+]
+
+# the es2015 filegroups
+[
+    filegroup(
+        name = b + ".umd",
+        srcs = [":" + b],
+        output_group = "umd",
+    )
+    for b in ES2015_BUNDLES
+]
+
+# the es2015.min filegroups
+[
+    filegroup(
+        name = b + ".umd.min",
+        srcs = [":" + b],
+        output_group = "umd_min",
+    )
+    for b in ES2015_BUNDLES
+]
+
+# Extract and rename each es5 bundle to a .js and .min.js in the dist/ dir
```

Comment:
this seems wrong. If we want the artifacts to appear in the `dist/` subfolder in the output tree (and therefore in the npm package) then we should probably have a BUILD file in `dist/` in the source tree which declares them. then we don't have to move things around. I can do some refactoring to help with that.

Reply:
Got it, I made it work, and testing it now.